src/backend/distributed/metadata/metadata_utility.c

```diff
@@ -628,6 +634,39 @@ GenerateSizeQueryOnMultiplePlacements(List *shardIntervalList, char *sizeQuery)
 }
 
 
+/*
+ * GetSizeQueryBySizeQueryType returns the corresponding size query for given query type.
+ * Errors out for an invalid query type.
+ */
+static char *
+GetSizeQueryBySizeQueryType(SizeQueryType sizeQueryType)
+{
+	switch (sizeQueryType)
+	{
+		case RELATION_SIZE:
+		{
+			return PG_RELATION_SIZE_FUNCTION;
+		}
+		case TOTAL_RELATION_SIZE:
+		{
+			return PG_TOTAL_RELATION_SIZE_FUNCTION;
+		}
+		case CSTORE_TABLE_SIZE:
+		{
+			return CSTORE_TABLE_SIZE_FUNCTION;
+		}
+		case TABLE_SIZE:
+		{
+			return PG_TABLE_SIZE_FUNCTION;
+		}
+		default:
+		{
+			elog(ERROR, "Size query type couldn't be found.");
+		}
+	}
+}
```

Comment:
Nice, was thinking about suggesting the extraction in a function, but didn't want to enforce it. You taking this initiative yourself is great!